internal/core/model.rs

```diff
@@ -503,18 +503,33 @@ impl Model for bool {
     }
 }
 
-/// A Reference counted [`Model`].
+/// ModelRc is a type alias for a reference counted implementation of the [`Model`] trait.
+///
+/// Models are used to represent sequences of the same data type. In `.slint` code those
+/// are represented using the `[T]` array syntax and typically used in `for` expressions,
+/// array properties, and array struct fields.
 ///
-/// The `ModelRc` struct holds something that implements the [`Model`] trait.
-/// This is used in `for` expressions in the .slint language.
-/// Array properties in the .slint language are holding a ModelRc.
 /// For example, a `property <[string]> foo` will be of type `ModelRc<SharedString>`
 /// and, behind the scenes, wraps a `Rc<dyn Model<Data = SharedString>>.`
 ///
-/// An empty model can be constructed with [`ModelRc::default()`].
-/// Use [`ModelRc::new()`] To construct a ModelRc from something that implements the
-/// [`Model`] trait.
-/// It is also possible to use the [`From`] trait to convert from `Rc<dyn Model>`.
+/// An array struct field will also be of type `ModelRc`:
+///
+/// ```slint,ignore
```

Comment:
Why ignore? Shouldn't we check the syntax of this?

Reply:
I suppose we should... I copied from somewhere else heh. I'll try to fix that. Maybe it's because it's only a snippet but it won't compile.